src/config.rs

```diff
@@ -5,13 +9,65 @@ pub trait Config {
 
     fn current_shell_name(&self) -> String;
 
-    fn set_current_shell_name(&mut self, name: &str);
+    fn set_current_shell_name(&mut self, name: &str) -> io::Result<()>;
 
-    fn does_shell_exist(&self, name: &str) -> bool;
+    fn does_shell_exist(&self, name: &str) -> io::Result<bool>;
```

Comment:
No.  Sorry for the confusion.  I like that pattern of instead of requiring consumers to handle a Result<bool>` type (which is awkward), of having the api do something like `<...>.unwrap_or(false)` as the `is_dir` method does.